### ISSUE TITLE
osm2pgsql stops with no errors or information

### ISSUE THREAD

**Issue by eleaj** [openstreetmap/osm2pgsql#575]
I have been trying to import an osm planet file in .pbf format into PostGIS and it appears to just stop after a few days of running. I've read through quite a few of the posts in the osm wiki forum and posted my own question and have got no answers on this issue. As far as I understand it, I should get some information about how long it has taken when it finishes but the last thing it shows me is something like this:

'Processing: Node<1929394k 225.9k/s> Way<186904k 1.37k/s> Relation<477740 9.65/s>'

I've tried it twice with the same file and a third time with a completely new planet file and all three times it does exactly the same thing. I checked the event logs each time and I can't see anything that says I've run out of memory or space and when I look in PostgreSQL there is only one table called spatial_ref_sys.

My server set up is thus:

Windows Server 2008 R2 - 64bit
2.00Ghz Dual Core
24 GB RAM
350GB HDD
The command I used to set osm2pgsql running was:

osm2pgsql -S "C:/Program Files/cygwin-package/default.style" -U postgres -d maps C:/Users/jordane/Desktop/Maps/planet-130614.osm.pbf -W --port 5432 --host localhost -s --flat-nodes flat-nodes.bin

Either I'm doing something wrong or there is an uncaught error/bug.

Thanks

**Comment by pnorman** [comment]
What version?

If it's got that far it's created the tables, so you must not be looking in the right database

**Comment by eleaj** [comment]
Which program do you need the version of and how do I find it?

I've only got two databases, both of which I have tried to import the data into using osm2pgsql and both only have the spatial_ref_sys table left in them. I may well be looking in the wrong place though, below is a screen shot of the two databases in question.

I monitored my disk space each time osm2pgsql was running and my disk would gradually fill up (still with plenty of available space), and then when the process stopped it would reduce back down again so I can only assume that it hits the point where it stops and then cleans up the partially imported data.

![image](https://cloud.githubusercontent.com/assets/18700431/14880316/170fde3a-0d26-11e6-8bfb-2870897a96a3.png)

**Comment by pnorman** [comment]
the osm2pgsql version is at the top of its output.

**Comment by ChiaChia77** [comment]
I have same issue:
`osm2pgsql -c -d osm -U osm -W -H 192.168.1.3 -C 30000 planet-latest.osm.pbf
osm2pgsql version 0.91.0-dev (64 bit id space)

Password:
Using built-in tag processing pipeline
Using projection SRS 3857 (Spherical Mercator)
Setting up table: planet_osm_point
Setting up table: planet_osm_line
Setting up table: planet_osm_polygon
Setting up table: planet_osm_roads
Allocating memory for dense node cache
Allocating dense node cache in one big chunk
Allocating memory for sparse node cache
Sharing dense sparse
Node-cache: cache=30000MB, maxblocks=480000*65536, allocation method=3
Mid: Ram, scale=100

Reading in file: planet-latest.osm.pbf
Using PBF parser.
Processing: Node(481590k 617.4k/s) Way(0k 0.00k/s) Relation(0 0.00/s)Killed
`
Tables is not create, data not import.

**Comment by pnorman** [comment]
> I have same issue:

No, this is a different error. In your case, something is killing osm2pgsql. You'd have to check your system logs to be sure, but odds are it's the out of memory killer.

**Comment by ChiaChia77** [comment]
Yes, I see it. Thank you.
```
kernel: pid 85210 (osm2pgsql), uid 0, was killed: out of swap space
```

**Comment by pnorman** [comment]
> What version?

The first line of the osm2pgsql output should have the information.

> 350GB HDD

looking at it, this is the probably the problem. Does it work with an extract?
